suricata/update/main.py

```diff
@@ -99,6 +99,11 @@
 # Rule keywords that come with files
 file_kw = ["filemd5", "filesha1", "filesha256", "dataset"]
 
+def emit(msg):
+    logger.error(msg)
+    if config.args().fail:
+        sys.exit(1)
+
```

Comment:
I think a better name for this is `strict_error` or something to signify that it logs an error and optionally exits.